Auth subject place access points, refs #12593

sbreker opened the pull request:
This is Feature 1 of a larger project.

sbreker committed:
Auth subject place access points, refs #12593

Add ability to have subject and place access points on authority
record.

sbreker committed:
Auth subject place ES indexing, refs #12593

Index place and subject access points with actors.

sbreker committed:
Auth subject place facets, refs #12593

Added subject and place facets on actor browse page.

sbreker committed:
Auth access points EAC XML, refs #12593

Logic to import and export EAC XML that includes the actor linked place
and subject access points.

sbreker committed:
Auth subject place terms CSV import, refs #12593

Added CSV import logic.

sbreker committed:
Auth place subjects csv export, refs #12593

Added CSV export logic.

sbreker committed:
Term browse page wording change, refs #12593

On the term browse page change the wording of the 'number of results'
message to include the fact that this is listing information objects
since this subject and place terms can now also be applied to actors.

sbreker committed:
Auth access points refactor, refs #12593

Moved the _placeAccessPoints.php and _subjectAccessPoints.php partials
to the object module since this is now called from the actor module too.

mcantelon commented on lib/model/QubitActor.php:
```diff
@@ -607,4 +607,20 @@ public function getOccupations()
+    return $this->getTermRelations(QubitTaxonomy::SUBJECT_ID);
+  }
+
+  public function getPlaceAccessPoints(/*array $options = array('events' => false)*/)
```
Ghost code. ;)

sbreker replied:
Removed this - see latest commit.

mcantelon commented on lib/model/QubitObjectTermRelation.php:
```diff
@@ -70,4 +70,40 @@ public static function getOneByObjectId($objectId)
+    $criteria = new Criteria;
+    $criteria->addJoin(QubitTerm::ID, QubitTermI18n::ID);
+    $criteria->add(QubitTerm::TAXONOMY_ID, $options['taxonomyId']);
+    $criteria->add(QubitTermI18n::NAME, $name);
```
Do we need to worry about culture?

sbreker replied:
I think you're right.  I have added a new commit that adds 'culture' to $options and updates the calls accordingly.

mcantelon reviewed (changes requested):
Looks good! One one minor thing found and a question.

sbreker committed:
Auth terms feedback changes, refs #12593

Term culture lookup and create:
- moved setTermRelationByName() to QubitActor and added 'culture'
  to the options array
- Updated calls for Actor EAC and CSV load to pass in culture from csv.

Removed commented out code.

mcantelon commented:
Looks good! Nice work! 👍
